[package] fix persistent_load error

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58439 [package] fix persistent_load error**

Differential Revision: [D28494250](https://our.internmc.facebook.com/intern/diff/D28494250)

Commit:
[package] fix persistent_load error

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28494250).

suo commented on torch/package/package_importer.py:
```diff
@@ -208,6 +208,9 @@ def persistent_load(saved_id):
                 storage = loaded_storages[key]
                 return storage
             elif typename == "reduce_package":
+                if len(data) == 2:
```
maybe add a comment that this is for BC, and causes produced objects not to be shared properly

Commit:
Update on "[package] fix persistent_load error"


Differential Revision: [D28494250](https://our.internmc.facebook.com/intern/diff/D28494250)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28494250).

Commit:
Update on "[package] fix persistent_load error"


Differential Revision: [D28494250](https://our.internmc.facebook.com/intern/diff/D28494250)

[ghstack-poisoned]

Lilyjjo commented:
@Lilyjjo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28494250).
